### PR TITLE
Force Node 16 usage in Manylinux CI

### PR DESCRIPTION
**Pull request by JakaMohorkoDS**
# Description:

Due to [GitHub actions transitioning to Node20](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/), the manylinux CI jobs are failing. This fix enforces the usage of Node16 in those actions.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -461,7 +462,7 @@ jobs:
         run : git config --global --add safe.directory '*'
 
       - name: Checkout
-        uses: actions/checkout@v3 # TODO https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/
+        uses: actions/checkout@v3
```
Comment (mojca) [inline .github/workflows/ci.yml:465]: Doesn't the following fix the original problem? (Please note that I did not test.)
```suggestion
        uses: actions/checkout@v4
```

Comment (mojca) [reply .github/workflows/ci.yml:465]: Ah, sorry. That's an ancient Linux distro that would probably require some tweaking to get Node 20 running on it in the first place.
Like https://tecadmin.net/install-latest-nodejs-and-npm-on-centos/ pointing to https://rpm.nodesource.com/

**Comment by aljazfrancic** [comment]
Not only has Node 16 reached EoL, but [so has CentOS](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/).
I would suggest foregoing CentOS in our CI.

**Comment by mojca** [comment]
> Not only has Node 16 reached EoL, but [so has CentOS](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/). I would suggest foregoing CentOS in our CI.

According to https://github.com/pypa/manylinux?tab=readme-ov-file#docker-images maybe the best way would be to select `manylinux_2_28` with GLIBC 2.28 (based on AlmaLinux 8 which is kind of "successor" to the discontinued CentOS).

**Review by aljazfrancic — APPROVED** [review]
`ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true` seems to do the trick for now (everything's green). [However, this will only work until they upgrade the runner, removing Node16 later in "the spring".](https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/) So I'd merge this so we can continue with other PRs, but would consider this as a temporary workaround.

**Comment by JakaMohorkoDS** [comment]
> > Not only has Node 16 reached EoL, but [so has CentOS](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/). I would suggest foregoing CentOS in our CI.
> 
> According to https://github.com/pypa/manylinux?tab=readme-ov-file#docker-images maybe the best way would be to select `manylinux_2_28` with GLIBC 2.28 (based on AlmaLinux 8 which is kind of "successor" to the discontinued CentOS).

Sure, but that means our binaries are compatible with less systems than before. I'll keep this workaround for now, and update to a newer manylinux version later.

**Pull request merged**